### PR TITLE
fix(core): add RTL to left and right placement popover and visual fixes

### PR DESCRIPTION
**Pull request by genereg1**
#### Please provide a link to the associated issue.
Closes https://github.com/SAP/fundamental-ngx/issues/6287

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [x] Run npm run build-pack-library and test in external application

Documentation checklist:
- [n/a] update `README.md`
- [n/a] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [n/a] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

**Comment by droshev** [comment]
@genereg1 can you rephrase the PR title? It doesn't say what it fixes

**Review by olshansky — APPROVED** [review]

File: libs/core/src/lib/shared/interfaces/popover-position.ts [hunk 1]
```diff
@@ -82,6 +82,15 @@ export const PopoverFlippedYDirection: {[key: string]: YPositions} = {
     'center': 'center'
 };
 
+export const PopoverFlippedRtlPlacement: {[key: string]: Placement} = {
```
Comment (dimamarksman) [inline libs/core/src/lib/shared/interfaces/popover-position.ts:85]: `{[key: Placement]: Placement}`?

File: libs/core/src/lib/shared/interfaces/popover-position.ts [hunk 2]
```diff
@@ -82,6 +82,15 @@ export const PopoverFlippedYDirection: {[key: string]: YPositions} = {
...
+    'left': 'right',
+    'left-end': 'right-end',
+    'right-start': 'left-start',
+    'right': 'left',
+    'right-end': 'left-end',
+}
+
 export class PopoverPosition {
 
     static getCdkPlacement(placement: Placement): ConnectedPosition {
```
Comment (dimamarksman) [inline libs/core/src/lib/shared/interfaces/popover-position.ts:96]: Can we extend this method adding `rtl = false` parameter, how it's done for `PopoverPosition .getArrowPosition`

So consumer doesn't need to be aware of Flipped version.

**Review by Betrozov — APPROVED** [review]

**Review by mikerodonnell89 — APPROVED** [review]

File: libs/core/src/lib/shared/interfaces/popover-position.ts [hunk 3]
```diff
@@ -82,15 +89,32 @@ export const PopoverFlippedYDirection: {[key: string]: YPositions} = {
...
         }
 
+        if (direction === 'rtl') {
+            const startPlaceStr = placement.split('-')[0];
+
+            if (startPlaceStr === 'left' || startPlaceStr === 'right') {
+                return popoverPlacementMap[PopoverFlippedRtlPlacement[placement]];
+            }
+        }
+
```
Comment (InnaAtanasova) [inline libs/core/src/lib/shared/interfaces/popover-position.ts:117]: You could've used replace method to flip the left/right :)
Then you wouldn't need the PopoverFlippedRtlPlacement const. But I accept this solution, looks good

File: libs/core/src/lib/popover/popover-body/popover-body.component.scss [hunk 4]
```diff
@@ -150,6 +151,54 @@ $fd-popover-background-color: #fff;
 			}
 		}
 	}
+
```
Comment (InnaAtanasova) [inline libs/core/src/lib/popover/popover-body/popover-body.component.scss:154]: Please create a mixin:
```
@mixin set-border-radius($position, $radius) {
		border-#{$position}-right-radius: $radius;
    	border-#{$position}-left-radius: $radius;
}
```

Then you can use it like this:
```
@include set-border-radius(top, $fd-popover-border-radius);
```

or even 

```
@mixin set-border-radius($position, $radius: $fd-popover-border-radius) {
		border-#{$position}-right-radius: $radius;
    	border-#{$position}-left-radius: $radius;
}
```

and then:
```
@include set-border-radius(top);
```

You will avoid a looot of code :)

**Review by InnaAtanasova — APPROVED** [review]
